scipy/stats/tests/test_distributions.py

```diff
@@ -3990,6 +3991,82 @@ def test_burr_nan_mean_var_9544(self):
         assert_(np.isfinite(e4))
 
 
+class TestStudentizedRange(object):
+    # Critical Q values were selected from this table:
+    # https://link.springer.com/chapter/10.1007/978-1-4613-9629-1_12
+    # For alpha=.05 and .01, and for each value of v=[1, 3, 10, 20],
+    # a Q was picked from each table for k=[2, 8, 14, 20].
+    ks = [2, 8, 14, 20]
+    vs = [1, 3, 10, 20]
+    alpha05 = [17.97, 45.40, 54.33, 59.56,
+               4.501, 8.853, 10.35, 11.24,
+               3.151, 5.305, 6.028, 6.467,
+               2.950, 4.768, 5.357, 5.714]
+    alpha01 = [90.03, 227.2, 271.8, 298.0,
+               8.261, 15.64, 18.22, 19.77,
+               4.482, 6.875, 7.712, 8.226,
+               4.024, 5.839, 6.450, 6.823]
+    ks_d = np.ravel(list((repeat(ks, 4))))
+    vs_d = np.ravel(list(zip(*(repeat(vs, 4)))))
+    p95 = [.95] * 16
+    p99 = [.99] * 16
+    sig_k_v_q = (list(zip(p95, ks_d, vs_d, alpha05))
+                 + list(zip(p99, ks_d, vs_d, alpha01)))
+
+    @pytest.mark.parametrize("significance, k, v,q", sig_k_v_q)
```

Comment:
```suggestion
    @pytest.mark.parametrize("significance, k, v, q", sig_k_v_q)
```

Reply:
Actually, don't parameterize this. Just vectorize it (calculate everything in one call to `studentized_range.cdf`)